lib/util/LocationStatusManager.js

```diff
@@ -59,6 +195,452 @@ class LocationStatusManager {
+                }
+                this._logger.debug(`Mongo source data from _locationStatusStore for 
+                    state machine init`, { key, sourceData, initialState });
+            }
+            this._logger.info('Initializing state machine', 
+                { key, service, location, initialState, sourceScheduledResume: sourceData.scheduledResume });
+            const stateMachineInstance = new ServiceLocationStateMachine(service, location, initialState);
+
+            if (initialState === SERVICE_STATES.SCHEDULED_RESUME && sourceData.scheduledResume) {
+                try {
+                    stateMachineInstance.scheduledResumeDate = new Date(sourceData.scheduledResume);
+                } catch (dateError) {
+                    this._logger.error('Invalid scheduledResume date format, cannot set on state machine',
+                        { key, service, location, dateValue: sourceData.scheduledResume, error: dateError.message });
+                    stateMachineInstance.state = SERVICE_STATES.PAUSED; 
+                    stateMachineInstance.scheduledResumeDate = null;
+                }
+            }
+            this._stateMachines.set(key, stateMachineInstance);
+        }
+
+        const stateMachineInstance = this._stateMachines.get(key);
+
+        if (this._serviceConfig[service] && !this._serviceConfig[service].isMongo) {
+            this._logger.debug('State machine is ZK-based. Refreshing/validating its state from ZK.', 
+                { key, currentSmState: stateMachineInstance.state, 
+                currentSmScheduledResume: stateMachineInstance.scheduledResumeDate });
+            try {
+                const currentZkData = await this._getZkDataForLocation(service, location);
+                let newSmState = SERVICE_STATES.RUNNING;
+                let newScheduledResumeDate = null;
+
+                if (currentZkData.paused) {
+                    newSmState = currentZkData.scheduledResume ?
+                        SERVICE_STATES.SCHEDULED_RESUME : SERVICE_STATES.PAUSED;
+                }
+                
+                if (newSmState === SERVICE_STATES.SCHEDULED_RESUME && currentZkData.scheduledResume) {
+                    try {
+                        newScheduledResumeDate = new Date(currentZkData.scheduledResume);
+                    } catch (dateError) {
+                        this._logger.error('Invalid scheduledResume date format from ZK during refresh', 
+                            { key, dateValue: currentZkData.scheduledResume, error: dateError.message });
+                        newSmState = SERVICE_STATES.PAUSED;
+                        newScheduledResumeDate = null;
+                    }
+                }
+
+                if (stateMachineInstance.state !== newSmState || 
+                    (stateMachineInstance.scheduledResumeDate ? 
+                        stateMachineInstance.scheduledResumeDate.toISOString() : null) !== 
```

Comment:
```suggestion
                    (stateMachineInstance.scheduledResumeDate?.toISOString?.() 
```

Also what if we get two `null` (or `undefined` with my suggestion) -> the condition will pass, is that what we want?